src/modules/workspaces/workspaces.usecase.ts

```diff
@@ -1553,59 +1573,71 @@ export class WorkspacesUsecases {
         updatedAt: new Date(),
       });
 
-      await this.workspaceRepository.addUserToWorkspace(workspaceUser);
+      await this.workspaceRepository.addUserToWorkspace(workspaceUser, {
+        transaction,
+      });
 
       await this.teamRepository.addUserToTeam(
         workspace.defaultTeamId,
         user.uuid,
+        { transaction },
       );
 
-      await this.workspaceRepository.createItem({
-        itemId: rootFolder.uuid,
-        workspaceId: workspace.id,
-        itemType: WorkspaceItemType.Folder,
-        context: WorkspaceItemContext.Drive,
-        createdBy: user.uuid,
-        createdAt: new Date(),
-        updatedAt: new Date(),
-      });
+      await this.workspaceRepository.createItem(
+        {
+          itemId: rootFolder.uuid,
+          workspaceId: workspace.id,
+          itemType: WorkspaceItemType.Folder,
+          context: WorkspaceItemContext.Drive,
+          createdBy: user.uuid,
+          createdAt: new Date(),
+          updatedAt: new Date(),
+        },
+        { transaction },
+      );
 
-      await this.workspaceRepository.deleteInviteBy({ id: invite.id });
+      await this.workspaceRepository.deleteInviteBy(
+        { id: invite.id },
+        { transaction },
+      );
+
+      await this.adjustOwnerStorage(workspace.id, invite.spaceLimit, 'DEDUCT', {
+        transaction,
+      });
 
-      await this.adjustOwnerStorage(workspace.id, invite.spaceLimit, 'DEDUCT');
+      await transaction.commit();
 
       return workspaceUser.toJSON();
     } catch (error) {
-      let finalMessage = 'There was a problem accepting this invite! ';
-      const rollbackError = await this.rollbackUserAddedToWorkspace(
-        user.uuid,
-        workspace,
-      );
-
-      finalMessage += rollbackError
-        ? rollbackError.message
-        : 'rollback applied successfully';
+      await transaction.rollback();
```

Comment:
Is there any possibility of setting an expiration for the transaction also, so just in case the connection is lost or everything fails, the database itself expires the lock?

Reply:
I don't believe it's possible to do so in a sequelize config or inside a sequelize transaction, but it appears it can be configured in the db settings: https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-IDLE-IN-TRANSACTION-SESSION-TIMEOUT

Reply:
Add the flag then